neuraxle/steps/util.py

```diff
@@ -316,12 +318,89 @@ class DataShuffler:
 
 
 class NullPipelineSaver(PipelineSaver):
+    def load(self, pipeline: 'Pipeline', data_container: DataContainer) -> 'Pipeline':
+        return pipeline
+
     def can_load(self, pipeline: 'Pipeline', data_container: DataContainer) -> bool:
         return True
 
     def save(self, pipeline: 'Pipeline', data_container: DataContainer) -> 'Pipeline':
         return pipeline
 
+
+class ForEachDataInputs(TruncableSteps):
```

Comment:
It should not be a truncable step, because if this is a truncable step you might as well just use a Sequential Pipeline with batch_size = 1 and with that, the handle methods will be implemented such that the whole thing be better. 

Here, we simply want to inherit from MetaStepMixin.

Reply:
Also note that we might want to implement the handle_* methods here to create sub-data containers (e.g.: if the original data container was 3D, we'll want here to create many little 2D data containers and loop on them. 

Then, after having transformed the mini 2D data containers, we'll need a method to merge them back into a big 3D data container again. 

Note: these mechanics of having to create a sub-data container might be found in streaming pipelines or into other staps later on. We would already need to implement that for the StepClonerForEachDataInputs. So I suggest that we add an issue which is about having a method in a data container to iterate on it but not on the data, but rather on mini-data containers of itself, and also a static method to do like `DataContainer.from_subcontainers(subcontainers: List[DataContainer])`. 

For now just changing the class we inherit from is fine and we'll add the handle methods later. Please create the issue before resolving.

Reply:
we might need this later. well see resolving for now.

Reply:
#98